Improve clarity of cryptographically random number functions

darinadler opened the pull request:
#### 017574a4e19bb7dfb6793c13510f30020aad2cdf
<pre>
Improve clarity of cryptographically random number functions
<a href="https://bugs.webkit.org/show_bug.cgi?id=247581">https://bugs.webkit.org/show_bug.cgi?id=247581</a>
rdar://problem/102057413

Reviewed by Don Olmstead.

* Source/JavaScriptCore/assembler/AbstractMacroAssembler.cpp:
(JSC::AbstractMacroAssemblerBase::initializeRandom): Use cryptographicallyRandomNumber&lt;uint32_t&gt;.

* Source/JavaScriptCore/assembler/AbstractMacroAssembler.h: Remove unneeded include.

* Source/JavaScriptCore/heap/Heap.cpp:
(JSC::Heap::runBeginPhase): Use cryptographicallyRandomNumber&lt;uint32_t&gt;.

* Source/JavaScriptCore/heap/HeapInlines.h: Remove unneeded include.

* Source/JavaScriptCore/inspector/remote/socket/RemoteInspectorSocketEndpoint.cpp:
(Inspector::RemoteInspectorSocketEndpoint::generateConnectionID): Use
cryptographicallyRandomNumber&lt;ConnectionID&gt;. Also fix mistake where this was checking for the
empty value, but not the deleted value.

* Source/JavaScriptCore/jit/ExecutableAllocator.cpp: Use cryptographicallyRandomNumber&lt;uint32_t&gt;.

* Source/JavaScriptCore/runtime/JSGlobalObject.cpp:
(JSC::JSGlobalObject::JSGlobalObject): Use cryptographicallyRandomNumber&lt;uint32_t&gt;.
* Source/JavaScriptCore/runtime/VM.cpp:
(JSC::VM::VM): Ditto.

* Source/WTF/WTF.xcodeproj/project.pbxproj: Rename RandomNumber.cpp/h to WeakRandomNumber.cpp/h.
Remove RandomNumberSeed.h.
* Source/WTF/wtf/CMakeLists.txt: Ditto.

* Source/WTF/wtf/CryptographicallyRandomNumber.cpp:
(WTF::cryptographicallyRandomNumber&lt;uint32_t&gt;): Renamed from cryptographicallyRandomNumber.
(WTF::cryptographicallyRandomNumber&lt;uint64_t&gt;): Added. Moved here from RandomNumber.cpp.
(WTF::cryptographicallyRandomUnitInterval): Added. Moved here from RandomNumber.cpp.

* Source/WTF/wtf/CryptographicallyRandomNumber.h: Moved all the cryptographically random
functions here from RandomNumber.h. Use template argument for the return type of
cryptographicallyRandomNumber.

* Source/WTF/wtf/FileSystem.cpp:
(WTF::FileSystemImpl::makeSalt): Use cryptographicallyRandomValues.

* Source/WTF/wtf/HashTable.h: Use weakRandomNumber&lt;uint32_t&gt;.

* Source/WTF/wtf/LoggerHelper.h: Use cryptographicallyRandomNumber&lt;uint64_t&gt;.

* Source/WTF/wtf/RandomNumberSeed.h: Removed.

* Source/WTF/wtf/RobinHoodHashTable.h:
(WTF::RobinHoodHashTable::random): Use weakRandomNumber&lt;uint32_t&gt;.

* Source/WTF/wtf/Threading.cpp:
(WTF::initialize): Removed call to initializeRandomNumberGenerator, since we don&apos;t use either of
the random number generators in WebKit any more that require initialization.

* Source/WTF/wtf/WeakRandom.h: Use cryptographicallyRandomNumber&lt;unsigned&gt;.

* Source/WTF/wtf/WeakRandomNumber.cpp: Renamed from RandomNumber.cpp.
(WTF::weakRandomNumber&lt;unsigned&gt;): Renamed from weakRandomUint32.
(WTF::cryptographicallyRandomUnitInterval): Deleted. Moved to
CryptographicallyRandomNumber.cpp.
(WTF::cryptographicallyRandomUint32): Ditto.
(WTF::cryptographicallyRandomUint64): Ditto.

* Source/WTF/wtf/WeakRandomNumber.h: Renamed from RandomNumber.h.
Removed other functions. Renamed weakRandomUint32 to weakRandomNumber&lt;uint32_t&gt; to match the name
of the cryptographicallyRandomNumber function.

* Source/WebCore/Modules/cache/CacheStorageConnection.cpp: Updated include.

* Source/WebCore/Modules/mediasession/MediaSession.cpp:
(WebCore::nextLogIdentifier): Use cryptographicallyRandomNumber&lt;uint32_t&gt;().
* Source/WebCore/Modules/mediasession/MediaSessionCoordinator.cpp:
(WebCore::nextCoordinatorLogIdentifier): Ditto.

* Source/WebCore/Modules/mediastream/MediaDevices.cpp: Updated include.

* Source/WebCore/Modules/mediastream/RTCPeerConnection.cpp:
(WebCore::RTCPeerConnection::RTCPeerConnection): Use LoggerHelper::uniqueLogIdentifier
instead of replicating what it does here.

* Source/WebCore/Modules/speech/SpeechRecognitionCaptureSourceImpl.cpp:
(WebCore::nextLogIdentifier): Use cryptographicallyRandomNumber&lt;uint32_t&gt;.

* Source/WebCore/dom/Document.cpp:
(WebCore::sharedLoggerOwner): Use cryptographicallyRandomNumber&lt;uint32_t&gt;.

* Source/WebCore/loader/PrivateClickMeasurement.cpp: Updated include.
* Source/WebCore/page/DOMTimer.cpp: Ditto.
* Source/WebCore/page/DiagnosticLoggingClient.h: Ditto.

* Source/WebCore/platform/graphics/TrackBuffer.cpp:
(WebCore::TrackBuffer::setLogger): Use cryptographicallyRandomNumber&lt;uint32_t&gt;.

* Source/WebCore/platform/mediastream/RealtimeMediaSourceCenter.cpp: Removed unneeded include.
* Source/WebCore/platform/mediastream/RealtimeOutgoingAudioSource.cpp: Ditto.

* Source/WebCore/platform/mediastream/gstreamer/GStreamerDisplayCaptureDeviceManager.cpp:
(WebCore::GStreamerDisplayCaptureDeviceManager::createDisplayCaptureSource): Use
weakRandomNumber&lt;uint32_t&gt;.

* Source/WebCore/platform/network/FormDataBuilder.cpp:
(WebCore::FormDataBuilder::generateUniqueBoundaryString): Use
cryptographicallyRandomNumber&lt;unsigned&gt;.

* Source/WebCore/style/RuleSetBuilder.cpp:
(WebCore::Style::RuleSetBuilder::pushCascadeLayer): Use cryptographicallyRandomNumber&lt;uint64_t&gt;.

* Source/WebKit/NetworkProcess/NetworkProcess.cpp:
(WebKit::NetworkProcess::randomClosedPortDelay): Use cryptographicallyRandomUnitInterval.

* Source/WebKit/NetworkProcess/NetworkSocketStream.cpp: Removed unneeded include.

* Source/WebKit/NetworkProcess/PrivateClickMeasurement/PrivateClickMeasurementManager.cpp:
(WebKit::PrivateClickMeasurementManager::randomlyBetweenFifteenAndThirtyMinutes const):
Use cryptographicallyRandomNumber&lt;uint32_t&gt;.

* Source/WebKit/NetworkProcess/cache/NetworkCacheData.cpp: Removed unneeded include.

* Source/WebKit/NetworkProcess/cache/NetworkCacheStorage.cpp: Updated include.

* Source/WebKit/Platform/IPC/win/ConnectionWin.cpp: Use cryptographicallyRandomNumber&lt;unsigned&gt;.
* Source/WebKit/Platform/unix/SharedMemoryUnix.cpp: Ditto.

* Source/WebKit/Shared/mac/AuxiliaryProcessMac.mm: Removed unneeded include.

* Source/WebKit/Shared/mac/MediaFormatReader/MediaFormatReader.cpp:
(WebKit::MediaFormatReader::nextLogIdentifier): Use cryptographicallyRandomNumber&lt;uint32_t&gt;.

* Source/WebKit/UIProcess/DeviceIdHashSaltStorage.cpp:
(WebKit::DeviceIdHashSaltStorage::completeDeviceIdHashSaltForOriginCall): Removed unneeded type
cast from call to cryptographicallyRandomValues.

* Source/WebKit/UIProcess/WebAuthentication/fido/CtapHidDriver.cpp:
(WebKit::CtapHidDriver::transact): Use weakRandomNumber&lt;uint32_t&gt;.

* Tools/Scripts/do-webcore-rename: Use to rename the files.

* Tools/TestWebKitAPI/Tests/WTF/BloomFilter.cpp:
(TestWebKitAPI::generateRandomHashes): Use cryptographicallyRandomNumber&lt;uint32_t&gt;.

* Tools/TestWebKitAPI/Tests/WTF/StdLibExtras.cpp: Updated include.

* Tools/TestWebKitAPI/Tests/WebKitCocoa/_WKWebAuthenticationPanel.mm:
(-[TestWebAuthenticationPanelDelegate panel:selectAssertionResponse:source:completionHandler:]):
Use weakRandomNumber&lt;uint32_t&gt;.

* Tools/WebKitTestRunner/TestController.cpp:
(WTR::TestController::configureWebsiteDataStoreTemporaryDirectories): Use
cryptographicallyRandomNumber&lt;uint32_t&gt;.

Canonical link: <a href="https://commits.webkit.org/256493@main">https://commits.webkit.org/256493@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0f7ebf4ff1073536cbbcfd2b7d4db6cd224db03b

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/95872 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/5119 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/28913 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/105433 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/165749 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/99854 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/5216 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/33881 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/88246 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/101264 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/101533 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/3840 "Found 2 new test failures: fast/images/text-recognition/ios/show-data-detector-context-menu.html, system-preview/badge.html (failure)") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/82479 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/30883 "Passed tests") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/85697 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/87603 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/73716 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/86904 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/39613 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/19139 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/82240 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/37295 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/20471 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/28213 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/4488 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/41460 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/43098 "Passed tests") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/84917 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/43482 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/39723 "Passed tests") | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/19179 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/JavaScriptCore/inspector/remote/socket/RemoteInspectorSocketEndpoint.cpp:
```diff
@@ -191,12 +191,10 @@ void RemoteInspectorSocketEndpoint::workerThread()
-    } while (!id || m_clients.contains(id) || m_listeners.contains(id));
-
+        id = cryptographicallyRandomNumber<ConnectionID>();
+    } while (!m_clients.isValidKey(id) || m_clients.contains(id) || m_listeners.contains(id));
```
This is a fix where we now correctly check for `0xFFFFFFFF` as well as for `0`.

darinadler commented on Source/WebCore/Modules/mediasession/MediaSession.cpp:
```diff
@@ -41,14 +41,15 @@
 static const void* nextLogIdentifier()
 {
-    static uint64_t logIdentifier = cryptographicallyRandomNumber();
+    static uint64_t logIdentifier = cryptographicallyRandomNumber<uint32_t>();
```
Left this as is, generating a 32-bit random number and storing it in a 64-bit global. New function name makes that a lot more obvious. Not sure if it’s a mistake.

darinadler commented on Source/WebCore/Modules/mediasession/MediaSessionCoordinator.cpp:
```diff
@@ -47,7 +48,7 @@ namespace WebCore {
 static const void* nextCoordinatorLogIdentifier()
 {
-    static uint64_t logIdentifier = cryptographicallyRandomNumber();
+    static uint64_t logIdentifier = cryptographicallyRandomNumber<uint32_t>();
```
Left this as is, generating a 32-bit random number and storing it in a 64-bit global. New function name makes that a lot more obvious. Not sure if it’s a mistake.

darinadler commented on Source/WebCore/Modules/speech/SpeechRecognitionCaptureSourceImpl.cpp:
```diff
@@ -40,7 +41,7 @@ namespace WebCore {
 static const void* nextLogIdentifier()
 {
-    static uint64_t logIdentifier = cryptographicallyRandomNumber();
+    static uint64_t logIdentifier = cryptographicallyRandomNumber<uint32_t>();
```
Left this as is, generating a 32-bit random number and storing it in a 64-bit global. New function name makes that a lot more obvious. Not sure if it’s a mistake.

darinadler commented on Source/WebCore/dom/Document.cpp:
```diff
@@ -460,7 +461,7 @@ uint64_t Document::s_globalTreeVersion = 0;
 static const void* sharedLoggerOwner()
 {
-    static uint64_t owner = cryptographicallyRandomNumber();
+    static uint64_t owner = cryptographicallyRandomNumber<uint32_t>();
```
Left this as is, generating a 32-bit random number and storing it in a 64-bit global. New function name makes that a lot more obvious. Not sure if it’s a mistake.

darinadler commented on Source/WebKit/Shared/mac/MediaFormatReader/MediaFormatReader.cpp:
```diff
@@ -296,7 +297,7 @@ const void* MediaFormatReader::nextTrackReaderLogIdentifier(uint64_t trackID) co
 const void* MediaFormatReader::nextLogIdentifier()
 {
-    static uint64_t logIdentifier = cryptographicallyRandomNumber();
+    static uint64_t logIdentifier = cryptographicallyRandomNumber<uint32_t>();
```
Left this as is, generating a 32-bit random number and storing it in a 64-bit global. New function name makes that a lot more obvious. Not sure if it’s a mistake.

donny-dont reviewed (approved):
r=me Great refactor!

Did you also audit the uses here or are we still doing that in another bug? The use of a cryptographically random log identifier seems wrong to me not like there'd be much of a performance hit since they're static. I would also guess they're being initialized with a 32-bit number just in case of overflow since they get incremented each call to the function.

darinadler committed:
Improve clarity of cryptographically random number functions
https://bugs.webkit.org/show_bug.cgi?id=247581
rdar://problem/102057413

Reviewed by Don Olmstead.

* Source/JavaScriptCore/assembler/AbstractMacroAssembler.cpp:
(JSC::AbstractMacroAssemblerBase::initializeRandom): Use cryptographicallyRandomNumber<uint32_t>.

* Source/JavaScriptCore/assembler/AbstractMacroAssembler.h: Remove unneeded include.

* Source/JavaScriptCore/heap/Heap.cpp:
(JSC::Heap::runBeginPhase): Use cryptographicallyRandomNumber<uint32_t>.

* Source/JavaScriptCore/heap/HeapInlines.h: Remove unneeded include.

* Source/JavaScriptCore/inspector/remote/socket/RemoteInspectorSocketEndpoint.cpp:
(Inspector::RemoteInspectorSocketEndpoint::generateConnectionID): Use
cryptographicallyRandomNumber<ConnectionID>. Also fix mistake where this was checking for the
empty value, but not the deleted value.

* Source/JavaScriptCore/jit/ExecutableAllocator.cpp: Use cryptographicallyRandomNumber<uint32_t>.

* Source/JavaScriptCore/runtime/JSGlobalObject.cpp:
(JSC::JSGlobalObject::JSGlobalObject): Use cryptographicallyRandomNumber<uint32_t>.
* Source/JavaScriptCore/runtime/VM.cpp:
(JSC::VM::VM): Ditto.

* Source/WTF/WTF.xcodeproj/project.pbxproj: Rename RandomNumber.cpp/h to WeakRandomNumber.cpp/h.
Remove RandomNumberSeed.h.
* Source/WTF/wtf/CMakeLists.txt: Ditto.

* Source/WTF/wtf/CryptographicallyRandomNumber.cpp:
(WTF::cryptographicallyRandomNumber<uint32_t>): Renamed from cryptographicallyRandomNumber.
(WTF::cryptographicallyRandomNumber<uint64_t>): Added. Moved here from RandomNumber.cpp.
(WTF::cryptographicallyRandomUnitInterval): Added. Moved here from RandomNumber.cpp.

* Source/WTF/wtf/CryptographicallyRandomNumber.h: Moved all the cryptographically random
functions here from RandomNumber.h. Use template argument for the return type of
cryptographicallyRandomNumber.

* Source/WTF/wtf/FileSystem.cpp:
(WTF::FileSystemImpl::makeSalt): Use cryptographicallyRandomValues.

* Source/WTF/wtf/HashTable.h: Use weakRandomNumber<uint32_t>.

* Source/WTF/wtf/LoggerHelper.h: Use cryptographicallyRandomNumber<uint64_t>.

* Source/WTF/wtf/RandomNumberSeed.h: Removed.

* Source/WTF/wtf/RobinHoodHashTable.h:
(WTF::RobinHoodHashTable::random): Use weakRandomNumber<uint32_t>.

* Source/WTF/wtf/Threading.cpp:
(WTF::initialize): Removed call to initializeRandomNumberGenerator, since we don't use either of
the random number generators in WebKit any more that require initialization.

* Source/WTF/wtf/WeakRandom.h: Use cryptographicallyRandomNumber<unsigned>.

* Source/WTF/wtf/WeakRandomNumber.cpp: Renamed from RandomNumber.cpp.
(WTF::weakRandomNumber<unsigned>): Renamed from weakRandomUint32.
(WTF::cryptographicallyRandomUnitInterval): Deleted. Moved to
CryptographicallyRandomNumber.cpp.
(WTF::cryptographicallyRandomUint32): Ditto.
(WTF::cryptographicallyRandomUint64): Ditto.

* Source/WTF/wtf/WeakRandomNumber.h: Renamed from RandomNumber.h.
Removed other functions. Renamed weakRandomUint32 to weakRandomNumber<uint32_t> to match the name
of the cryptographicallyRandomNumber function.

* Source/WebCore/Modules/cache/CacheStorageConnection.cpp: Updated include.

* Source/WebCore/Modules/mediasession/MediaSession.cpp:
(WebCore::nextLogIdentifier): Use cryptographicallyRandomNumber<uint32_t>().
* Source/WebCore/Modules/mediasession/MediaSessionCoordinator.cpp:
(WebCore::nextCoordinatorLogIdentifier): Ditto.

* Source/WebCore/Modules/mediastream/MediaDevices.cpp: Updated include.

* Source/WebCore/Modules/mediastream/RTCPeerConnection.cpp:
(WebCore::RTCPeerConnection::RTCPeerConnection): Use LoggerHelper::uniqueLogIdentifier
instead of replicating what it does here.

* Source/WebCore/Modules/speech/SpeechRecognitionCaptureSourceImpl.cpp:
(WebCore::nextLogIdentifier): Use cryptographicallyRandomNumber<uint32_t>.

* Source/WebCore/dom/Document.cpp:
(WebCore::sharedLoggerOwner): Use cryptographicallyRandomNumber<uint32_t>.

* Source/WebCore/loader/PrivateClickMeasurement.cpp: Updated include.
* Source/WebCore/page/DOMTimer.cpp: Ditto.
* Source/WebCore/page/DiagnosticLoggingClient.h: Ditto.

* Source/WebCore/platform/graphics/TrackBuffer.cpp:
(WebCore::TrackBuffer::setLogger): Use cryptographicallyRandomNumber<uint32_t>.

* Source/WebCore/platform/mediastream/RealtimeMediaSourceCenter.cpp: Removed unneeded include.
* Source/WebCore/platform/mediastream/RealtimeOutgoingAudioSource.cpp: Ditto.

* Source/WebCore/platform/mediastream/gstreamer/GStreamerDisplayCaptureDeviceManager.cpp:
(WebCore::GStreamerDisplayCaptureDeviceManager::createDisplayCaptureSource): Use
weakRandomNumber<uint32_t>.

* Source/WebCore/platform/network/FormDataBuilder.cpp:
(WebCore::FormDataBuilder::generateUniqueBoundaryString): Use
cryptographicallyRandomNumber<unsigned>.

* Source/WebCore/style/RuleSetBuilder.cpp:
(WebCore::Style::RuleSetBuilder::pushCascadeLayer): Use cryptographicallyRandomNumber<uint64_t>.

* Source/WebKit/NetworkProcess/NetworkProcess.cpp:
(WebKit::NetworkProcess::randomClosedPortDelay): Use cryptographicallyRandomUnitInterval.

* Source/WebKit/NetworkProcess/NetworkSocketStream.cpp: Removed unneeded include.

* Source/WebKit/NetworkProcess/PrivateClickMeasurement/PrivateClickMeasurementManager.cpp:
(WebKit::PrivateClickMeasurementManager::randomlyBetweenFifteenAndThirtyMinutes const):
Use cryptographicallyRandomNumber<uint32_t>.

* Source/WebKit/NetworkProcess/cache/NetworkCacheData.cpp: Removed unneeded include.

* Source/WebKit/NetworkProcess/cache/NetworkCacheStorage.cpp: Updated include.

* Source/WebKit/Platform/IPC/win/ConnectionWin.cpp: Use cryptographicallyRandomNumber<unsigned>.
* Source/WebKit/Platform/unix/SharedMemoryUnix.cpp: Ditto.

* Source/WebKit/Shared/mac/AuxiliaryProcessMac.mm: Removed unneeded include.

* Source/WebKit/Shared/mac/MediaFormatReader/MediaFormatReader.cpp:
(WebKit::MediaFormatReader::nextLogIdentifier): Use cryptographicallyRandomNumber<uint32_t>.

* Source/WebKit/UIProcess/DeviceIdHashSaltStorage.cpp:
(WebKit::DeviceIdHashSaltStorage::completeDeviceIdHashSaltForOriginCall): Removed unneeded type
cast from call to cryptographicallyRandomValues.

* Source/WebKit/UIProcess/WebAuthentication/fido/CtapHidDriver.cpp:
(WebKit::CtapHidDriver::transact): Use weakRandomNumber<uint32_t>.

* Tools/Scripts/do-webcore-rename: Use to rename the files.

* Tools/TestWebKitAPI/Tests/WTF/BloomFilter.cpp:
(TestWebKitAPI::generateRandomHashes): Use cryptographicallyRandomNumber<uint32_t>.

* Tools/TestWebKitAPI/Tests/WTF/StdLibExtras.cpp: Updated include.

* Tools/TestWebKitAPI/Tests/WebKitCocoa/_WKWebAuthenticationPanel.mm:
(-[TestWebAuthenticationPanelDelegate panel:selectAssertionResponse:source:completionHandler:]):
Use weakRandomNumber<uint32_t>.

* Tools/WebKitTestRunner/TestController.cpp:
(WTR::TestController::configureWebsiteDataStoreTemporaryDirectories): Use
cryptographicallyRandomNumber<uint32_t>.

Canonical link: https://commits.webkit.org/256493@main
